src/Reflection/ReflectionFunctionAbstract.php

```diff
@@ -270,17 +273,27 @@ public function isClosure() : bool
         return $this->node instanceof Node\Expr\Closure;
     }
 
-    /**
-     * Is this function deprecated?
-     *
-     * Note - we cannot reflect on internal functions (as there is no PHP source
-     * code we can access. This means, at present, we can only EVER return false
-     * from this function.
-     *
-     * @see https://github.com/Roave/BetterReflection/issues/38
-     */
     public function isDeprecated() : bool
     {
+        $docComment = $this->getDocComment();
+        if ($docComment !== '') {
+            $deprecatedTag = preg_match('#@deprecated\s+(\d+\.\d+(?:\.\d+)?)#', $docComment, $deprecatedTagMatches);
+            if ($deprecatedTag) {
+                return true;
```

Comment:
Do we have examples of functions for both of these paths? The `@deprecated` and the `#[Deprecated]` one?

Reply:
I will check inside the JetBrains stubs. I don't think so, but maybe for some of the extensions.

Reply:
Mostly `@deprecated` is used for `define()`. The `yaf` extension stubs have `@deprecated` over attributes and so does the Zip extension stubs.

See https://github.com/JetBrains/phpstorm-stubs/blob/c6e098ed55bb1e560f576495583335039c754816/zip/zip.php#L1339

I'll expand the tests via the data provider to catch these and ensure the regex is working